gax/src/main/java/com/google/api/gax/batching/BatcherImpl.java

```diff
@@ -279,17 +294,57 @@ private void awaitAllOutstandingBatches() throws InterruptedException {
   /** {@inheritDoc} */
   @Override
   public void close() throws InterruptedException {
-    if (isClosed) {
-      return;
+    try {
+      closeAsync().get();
+    } catch (ExecutionException e) {
+      // Original stacktrace of a batching exception is not useful, so rethrow the error with
+      // the caller stacktrace
+      if (e.getCause() instanceof BatchingException) {
+        BatchingException cause = (BatchingException) e.getCause();
+        throw new BatchingException(cause.getMessage());
+      } else {
+        throw new IllegalStateException("unexpected error closing the batcher", e.getCause());
```

Comment:
This will loose the ExecutionException itself (with maybe some useful message there). I.e. why `e.getCause()` instead of just `e` as the second argument? Get cause will still be there as a "sub-exception".

Reply:
ExecutionException doesn't provide any useful info here and leaks implementation details. The caller doesn't care that this was implemented with a future, but they do care that one of their elements failed.